crates/sqlbuiltins/src/functions/scalars.rs

```diff
@@ -0,0 +1,870 @@
+
+use crate::builtins::BuiltinFunction;
+// mostly using a macro here to preserve the formatting.
+// rustfmt will otherwise compact the lines.
+macro_rules! make_const {
+    (
+        $var_name:ident,
+        name => $name:expr,
+        example => $example:expr,
+        description => $description:expr
+    ) => {
+        const $var_name: (&str, &str, &str) = ($name, $example, $description);
+    };
+}
+
+make_const!(
+    ABS,
+    name => "abs",
+    example => "abs(-1)",
+    description => "Compute the absolute value of a number."
+);
+make_const!(
+    ACOS,
+    name => "acos",
+    example => "acos(0.5)",
+    description => "Compute the inverse cosine (arc cosine) of a number."
+);
+
+make_const!(
+    ACOSH,
+    name => "acosh",
+    example => "acosh(1)",
+    description => "Compute the inverse hyperbolic cosine of a number."
+);
+make_const!(
+    ASIN,
+    name => "asin",
+    example => "asin(0.5)",
+    description => "Compute the inverse sine (arc sine) of a number."
+);
+make_const!(
+    ASINH,
+    name => "asinh",
+    example => "asinh(1)",
+    description => "Compute the inverse hyperbolic sine of a number."
+);
+make_const!(
+    ATAN,
+    name => "atan",
+    example => "atan(1)",
+    description => "Computes the arctangent of a number."
+);
+make_const!(
+    ATAN2,
+    name => "atan2",
+    example => "atan2(1, 1)",
+    description => "Computes the arctangent of y/x given y and x."
+);
+make_const!(
+    ATANH,
+    name => "atanh",
+    example => "atanh(0.5)",
+    description => "Compute the inverse hyperbolic tangent of a number."
+);
+make_const!(
+    CBRT,
+    name => "cbrt",
+    example => "cbrt(27)",
+    description => "Compute the cube root of a number."
+);
+make_const!(
+    CEIL,
+    name => "ceil",
+    example => "ceil(1.1)",
+    description => "Compute the smallest integer greater than or equal to a number."
+);
+make_const!(
+    COS,
+    name => "cos",
+    example => "cos(0)",
+    description => "Compute the cosine of a number."
+);
+make_const!(
+    COT,
+    name => "cot",
+    example => "cot(1)",
+    description => "Compute the cotangent of a number."
+);
+make_const!(
+    COSH,
+    name => "cosh",
+    example => "cosh(0)",
+    description => "Compute the hyperbolic cosine of a number."
+);
+make_const!(
+    DEGREES,
+    name => "degrees",
+    example => "degrees(1)",
+    description => "Converts an angle measured in radians to an approximately equivalent angle measured in degrees."
+);
+make_const!(
+    EXP,
+    name => "exp",
+    example => "exp(1)",
+    description => "Compute the base-e exponential of a number."
+);
+make_const!(
+    FACTORIAL,
+    name => "factorial",
+    example => "factorial(5)",
+    description => "Compute the factorial of a number."
+);
+make_const!(
+    FLOOR,
+    name => "floor",
+    example => "floor(1.1)",
+    description => "Compute the largest integer less than or equal to a number."
```

Comment:
I would not assume (though perhaps I should,) that "`number`" meant float.

Reply:
it's a number because we do implicit casting to floats. `floor(1)` is valid.